src/groovy/org/transmartproject/db/dataquery/highdim/snp_lz/SnpLzProbabilitiesProjection.groovy

```diff
@@ -48,12 +52,13 @@ class SnpLzProbabilitiesProjection implements CriteriaProjection<double[]> {
     }
 
     @Override
-    double[] doWithResult(Object o) {
+    Map<String, Double> doWithResult(Object o) {
         assert o instanceof SnpLzAllDataCell
-        char[] res = new double[3]
-        res[0] = ((SnpLzAllDataCell) o).probabilityA1A1
-        res[1] = ((SnpLzAllDataCell) o).probabilityA1A2
-        res[2] = ((SnpLzAllDataCell) o).probabilityA2A2
-        res
+        SnpLzAllDataCell cell = (SnpLzAllDataCell) o
+        return [
+                probabilityA1A1: cell.probabilityA1A1,
+                probabilityA1A2: cell.probabilityA1A2,
+                probabilityA2A2: cell.probabilityA2A2
+        ] as Map<String, Double>
```

Comment:
Better to use `ImmutableMap`. The `as` operator tends to be very inefficient in Groovy (though maybe it as a shortcut in this case).